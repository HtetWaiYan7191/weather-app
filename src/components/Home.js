import React from 'react';
import citiesData from '../redux/citiesData';
import CityCard from './CityCard';
import '../styles/Home.css';

function Home () {



  return (
    <div className='city-container grid grid-cols-3'>
        {citiesData.map((city) => <CityCard key={city.name} city={city}/>)}
    </div>
  )
}
export default Home;
