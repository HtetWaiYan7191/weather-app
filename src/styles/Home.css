.city-container .link-wrapper .city-card {
    padding: 25px 15px;
}
.city-container .link-wrapper:nth-child(odd) {
    background-color:#468bd0e3;
}

.city-container .link-wrapper:nth-child(even) {
    background-color:#2f85c7ef;
}

.weather-info li {
    font-family: 'Roboto Slab', serif;
    color: whitesmoke;
}
