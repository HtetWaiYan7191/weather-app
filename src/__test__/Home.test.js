import React from "react";
import renderer from 'react-test-renderer';
import { MemoryRouter } from "react-router-dom";
import Home from "../components/Home";
import { render } from "@testing-library/react";

describe('Home component', () => {
    test('Home component renders correctly', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <Home/>
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})
