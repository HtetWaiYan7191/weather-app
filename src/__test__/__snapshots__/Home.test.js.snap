// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Home component renders correctly 1`] = `
<div>
  <div
    className="search-input-container flex justify-center w-[100%] my-10"
  >
    <form
      className="flex justify-around w-[90%]"
      onSubmit={[Function]}
    >
      <input
        className=" outline-none border-b-2 text-white border-b-white bg-inherit placeholder:text-white  md:mb-0"
        onChange={[Function]}
        placeholder="Search City"
        value=""
      />
      <button
        className=" md:hidden block hover:bg-sky-700 text-white bg-sky-800 rounded-md py-1 px-3"
        onClick={[Function]}
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
  <div
    className="city-container grid grid-cols-2 md:grid-cols-5 border border-t-white"
  >
    <a
      className="link-wrapper"
      href="/details/Yangon/16.8053/96.1561"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Yangon
          </li>
          <li>
            Lat :
            16.8053
          </li>
          <li>
            Lon :
            96.1561
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/London/44.34/10.99"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            London
          </li>
          <li>
            Lat :
            44.34
          </li>
          <li>
            Lon :
            10.99
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/Paris/48.8589/2.32"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Paris
          </li>
          <li>
            Lat :
            48.8589
          </li>
          <li>
            Lon :
            2.32
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/Rome/41.8933/12.4829"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Rome
          </li>
          <li>
            Lat :
            41.8933
          </li>
          <li>
            Lon :
            12.4829
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/Spain/40.4167/-3.7036"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Spain
          </li>
          <li>
            Lat :
            40.4167
          </li>
          <li>
            Lon :
            -3.7036
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/Berlin/52.517/13.3889"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Berlin
          </li>
          <li>
            Lat :
            52.517
          </li>
          <li>
            Lon :
            13.3889
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/Tokyo/35.6767/139.764"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Tokyo
          </li>
          <li>
            Lat :
            35.6767
          </li>
          <li>
            Lon :
            139.764
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/Beijing/39.9062/116.3913"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Beijing
          </li>
          <li>
            Lat :
            39.9062
          </li>
          <li>
            Lon :
            116.3913
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/Singapore/1.2899/103.8519"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Singapore
          </li>
          <li>
            Lat :
            1.2899
          </li>
          <li>
            Lon :
            103.8519
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/Mumbai/19.0785/72.8782"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Mumbai
          </li>
          <li>
            Lat :
            19.0785
          </li>
          <li>
            Lon :
            72.8782
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/New York/39.6853/-93.9269"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            New York
          </li>
          <li>
            Lat :
            39.6853
          </li>
          <li>
            Lon :
            -93.9269
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/Seoul/37.5665/126.978"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Seoul
          </li>
          <li>
            Lat :
            37.5665
          </li>
          <li>
            Lon :
            126.978
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/Bangkok/13.75/100.5167"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Bangkok
          </li>
          <li>
            Lat :
            13.75
          </li>
          <li>
            Lon :
            100.5167
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/Rio/38.3/21.7833"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Rio
          </li>
          <li>
            Lat :
            38.3
          </li>
          <li>
            Lon :
            21.7833
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/Hong Kong/22.2855/114.1577"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Hong Kong
          </li>
          <li>
            Lat :
            22.2855
          </li>
          <li>
            Lon :
            114.1577
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/Moscow/55.7522/37.6156"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Moscow
          </li>
          <li>
            Lat :
            55.7522
          </li>
          <li>
            Lon :
            37.6156
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/Stockholm/59.3314/18.0594"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Stockholm
          </li>
          <li>
            Lat :
            59.3314
          </li>
          <li>
            Lon :
            18.0594
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/Abu Dhabi/24.4667/54.3667"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Abu Dhabi
          </li>
          <li>
            Lat :
            24.4667
          </li>
          <li>
            Lon :
            54.3667
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/Cairo/30.0626/31.2497"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Cairo
          </li>
          <li>
            Lat :
            30.0626
          </li>
          <li>
            Lon :
            31.2497
          </li>
        </ul>
      </div>
    </a>
    <a
      className="link-wrapper"
      href="/details/Amsterdam/52.374/4.8897"
      onClick={[Function]}
    >
      <div
        className="city-card flex justify-between items-center cursor-pointer"
      >
        <figure>
          <img
            alt="cloud"
            className="md:w-[100px]  w-[70px]"
            src="cloudImage.png"
          />
        </figure>
        <ul
          className="weather-info flex flex-col"
        >
          <li>
            Amsterdam
          </li>
          <li>
            Lat :
            52.374
          </li>
          <li>
            Lon :
            4.8897
          </li>
        </ul>
      </div>
    </a>
  </div>
</div>
`;
